### ISSUE TITLE
Sonar doesn't work with Gradle out-of-the-box

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster#8503]
##### **Overview of the issue**

I created an app with the following JDL:

```
application {
  config {
    applicationType monolith,
    baseName HealthPoints
    packageName com.okta.developer,
    authenticationType oauth2,
    prodDatabaseType postgresql,
    buildTool gradle,
    searchEngine elasticsearch,
    testFrameworks [protractor],
    clientFramework react,
    useSass true,
    enableTranslation true,
    nativeLanguage en,
    languages [en, es]
  }
}

// JDL definition for application 'TwentyOnePoints' generated with command 'jhipster export-jdl'

entity BloodPressure {
  timestamp ZonedDateTime required,
  systolic Integer required,
  diastolic Integer required
}
entity Weight {
  timestamp ZonedDateTime required,
  weight Double required
}
entity Points {
  date LocalDate required,
  exercise Integer,
  meals Integer,
  alcohol Integer,
  notes String maxlength(140)
}
entity Preferences {
  weeklyGoal Integer required min(10) max(21),
  weightUnits Units required
}
enum Units {
  KG,
  LB
}
relationship OneToOne {
  Preferences{user(login)} to User
}
relationship ManyToOne {
  BloodPressure{user(login)} to User,
  Weight{user(login)} to User,
  Points{user(login)} to User
}
paginate BloodPressure, Weight with infinite-scroll
paginate Points with pagination
```

Then I started all the Docker containers needed to run it with the prod profile.

```
docker-compose -f src/main/docker/elasticsearch.yml up -d
docker-compose -f src/main/docker/keycloak.yml up -d
docker-compose -f src/main/docker/postgresql.yml up -d
docker-compose -f src/main/docker/sonar.yml up -d
```

Then I ran the following command to execute all the tests and track it in Sonar.

```
./gradlew -Pprod clean test sonarqube --stacktrace
```

This results in the following stacktrace:

```
> Task :webpack_test

> health-points@0.0.0 webpack:test /Users/mraible/okta-react-native-spring-boot-example/jhipster-api
> npm run test


> health-points@0.0.0 test /Users/mraible/okta-react-native-spring-boot-example/jhipster-api
> npm run lint && jest --coverage --logHeapUsage -w=2 --config src/test/javascript/jest.conf.js


> health-points@0.0.0 lint /Users/mraible/okta-react-native-spring-boot-example/jhipster-api
> tslint --project tsconfig.json -e 'node_modules/**'

PASS src/test/javascript/spec/app/modules/administration/administration.reducer.spec.ts (74 MB heap size)
PASS src/test/javascript/spec/app/shared/layout/header/header.spec.tsx (73 MB heap size)
PASS src/test/javascript/spec/app/config/notification-middleware.spec.ts (86 MB heap size)
PASS src/test/javascript/spec/app/shared/auth/private-route.spec.tsx (85 MB heap size)
  ● Console

    console.error node_modules/prop-types/checkPropTypes.js:19
      Warning: Failed context type: The context `router` is marked as required in `Redirect`, but its value is `undefined`.
          in Redirect

PASS src/test/javascript/spec/app/shared/reducers/application-profile.spec.ts (75 MB heap size)
PASS src/test/javascript/spec/app/shared/reducers/user-management.spec.ts (77 MB heap size)
PASS src/test/javascript/spec/app/shared/reducers/authentication.spec.ts (82 MB heap size)
(node:49722) UnhandledPromiseRejectionWarning: [object Object]
(node:49722) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:49722) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
PASS src/test/javascript/spec/app/config/axios-interceptor.spec.ts (84 MB heap size)
PASS src/test/javascript/spec/app/shared/util/entity-utils.spec.ts (96 MB heap size)
PASS src/test/javascript/spec/app/shared/reducers/locale.spec.ts (94 MB heap size)
PASS src/test/javascript/spec/app/shared/error/error-boundary-route.spec.tsx (101 MB heap size)
PASS src/test/javascript/spec/app/shared/layout/header/menus/account.spec.tsx (104 MB heap size)
PASS src/test/javascript/spec/app/shared/error/error-boundary.spec.tsx (102 MB heap size)
--------------------------------------------|----------|----------|----------|----------|-------------------|
File                                        |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
--------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                   |    90.65 |    73.96 |    78.21 |    92.82 |                   |
 main/webapp/app/config                     |    92.73 |    69.57 |    92.31 |    92.23 |                   |
  axios-interceptor.ts                      |      100 |    66.67 |      100 |      100 |                16 |
  constants.ts                              |      100 |      100 |      100 |      100 |                   |
  notification-middleware.ts                |    90.41 |    69.84 |      100 |    89.55 |... 96,100,101,103 |
  translation.ts                            |     87.5 |      100 |        0 |     87.5 |                17 |
 main/webapp/app/modules/administration     |      100 |    96.67 |      100 |      100 |                   |
  administration.reducer.ts                 |      100 |    96.67 |      100 |      100 |                37 |
 main/webapp/app/shared/auth                |    81.08 |    77.78 |    83.33 |     87.5 |                   |
  private-route.tsx                         |    81.08 |    77.78 |    83.33 |     87.5 |       37,69,70,71 |
 main/webapp/app/shared/error               |      100 |    72.73 |      100 |      100 |                   |
  error-boundary-route.tsx                  |      100 |      100 |      100 |      100 |                   |
  error-boundary.tsx                        |      100 |    66.67 |      100 |      100 |             27,28 |
 main/webapp/app/shared/layout/header       |       85 |      100 |    33.33 |     93.1 |                   |
  header-components.tsx                     |    71.43 |      100 |        0 |      100 |                   |
  header.tsx                                |    92.31 |      100 |       60 |    89.47 |             34,47 |
 main/webapp/app/shared/layout/header/menus |    82.22 |    33.33 |       20 |    94.59 |                   |
  account.tsx                               |      100 |      100 |      100 |      100 |                   |
  admin.tsx                                 |    78.57 |        0 |        0 |      100 |             44,46 |
  entities.tsx                              |       80 |      100 |        0 |      100 |                   |
  index.ts                                  |      100 |      100 |      100 |      100 |                   |
  locale.tsx                                |    55.56 |        0 |        0 |    71.43 |               7,9 |
 main/webapp/app/shared/reducers            |    94.25 |    77.42 |    94.74 |     94.2 |                   |
  action-type.util.ts                       |      100 |      100 |      100 |      100 |                   |
  application-profile.ts                    |      100 |       75 |      100 |      100 |                17 |
  authentication.ts                         |    96.67 |    83.33 |      100 |    95.65 |                29 |
  locale.ts                                 |    95.24 |    83.33 |      100 |    94.12 |                32 |
  user-management.ts                        |    83.33 |    66.67 |    66.67 |     87.5 |             23,37 |
 main/webapp/app/shared/util                |    92.86 |    71.43 |      100 |    90.91 |                   |
  entity-utils.ts                           |      100 |      100 |      100 |      100 |                   |
  url-utils.ts                              |       80 |       50 |      100 |       80 |                 4 |
 test/javascript/spec                       |    66.67 |        0 |       20 |    66.67 |                   |
  enzyme-setup.ts                           |      100 |      100 |      100 |      100 |                   |
  storage-mock.ts                           |    55.56 |        0 |       20 |    55.56 |           4,5,6,7 |
--------------------------------------------|----------|----------|----------|----------|-------------------|

Test Suites: 13 passed, 13 total
Tests:       78 passed, 78 total
Snapshots:   2 passed, 2 total
Time:        4.048s, estimated 5s
Ran all test suites.

> Task :sonarqube
Classes not found during the analysis : [com.datastax.driver.core.Cluster, org.openid4java.consumer.ConsumerException, org.openid4java.consumer.ConsumerManager, org.springframework.amqp.rabbit.connection.ConnectionFactory, org.springframework.data.mongodb.MongoDbFactory, org.springframework.data.redis.connection.RedisConnectionFactory, org.springframework.security.oauth2.client.OAuth2AuthorizedClientService, org.springframework.security.oauth2.client.registration.ClientRegistrationRepository, org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter, org.springframework.security.openid.AxFetchListFactory, org.springframework.security.openid.OpenIDAuthenticationFilter, org.springframework.security.openid.OpenIDAuthenticationToken, org.springframework.security.openid.OpenIDConsumer]
Property 'sonar.genericcoverage.unitTestReportPaths' is deprecated. Please use 'sonar.testExecutionReportPaths' instead.
Missing blame information for the following files:
  * src/main/java/com/okta/developer/ApplicationWebXml.java
  * src/main/java/com/okta/developer/HealthPointsApp.java
  * src/main/java/com/okta/developer/aop/logging/LoggingAspect.java
  * src/main/java/com/okta/developer/config/ApplicationProperties.java
  * src/main/java/com/okta/developer/config/AsyncConfiguration.java
  * src/main/java/com/okta/developer/config/CacheConfiguration.java
  * src/main/java/com/okta/developer/config/CloudDatabaseConfiguration.java
  * src/main/java/com/okta/developer/config/Constants.java
  * src/main/java/com/okta/developer/config/DatabaseConfiguration.java
  * src/main/java/com/okta/developer/config/DateTimeFormatConfiguration.java
  * src/main/java/com/okta/developer/config/DefaultProfileUtil.java
  * src/main/java/com/okta/developer/config/ElasticsearchConfiguration.java
  * src/main/java/com/okta/developer/config/JacksonConfiguration.java
  * src/main/java/com/okta/developer/config/LiquibaseConfiguration.java
  * src/main/java/com/okta/developer/config/LocaleConfiguration.java
  * src/main/java/com/okta/developer/config/LoggingAspectConfiguration.java
  * src/main/java/com/okta/developer/config/LoggingConfiguration.java
  * src/main/java/com/okta/developer/config/MetricsConfiguration.java
  * src/main/java/com/okta/developer/config/OAuth2Configuration.java
  * src/main/java/com/okta/developer/config/SecurityConfiguration.java
  * src/main/java/com/okta/developer/config/WebConfigurer.java
  * src/main/java/com/okta/developer/config/audit/AuditEventConverter.java
  * src/main/java/com/okta/developer/config/audit/package-info.java
  * src/main/java/com/okta/developer/config/package-info.java
  * src/main/java/com/okta/developer/domain/AbstractAuditingEntity.java
  * src/main/java/com/okta/developer/domain/Authority.java
  * src/main/java/com/okta/developer/domain/PersistentAuditEvent.java
  * src/main/java/com/okta/developer/domain/User.java
  * src/main/java/com/okta/developer/domain/package-info.java
  * src/main/java/com/okta/developer/repository/AuthorityRepository.java
  * src/main/java/com/okta/developer/repository/CustomAuditEventRepository.java
  * src/main/java/com/okta/developer/repository/PersistenceAuditEventRepository.java
  * src/main/java/com/okta/developer/repository/UserRepository.java
  * src/main/java/com/okta/developer/repository/package-info.java
  * src/main/java/com/okta/developer/repository/search/UserSearchRepository.java
  * src/main/java/com/okta/developer/repository/search/package-info.java
  * src/main/java/com/okta/developer/security/AuthoritiesConstants.java
  * src/main/java/com/okta/developer/security/OAuth2AuthenticationSuccessHandler.java
  * src/main/java/com/okta/developer/security/SecurityUtils.java
  * src/main/java/com/okta/developer/security/SpringSecurityAuditorAware.java
  * src/main/java/com/okta/developer/security/package-info.java
  * src/main/java/com/okta/developer/service/AuditEventService.java
  * src/main/java/com/okta/developer/service/UserService.java
  * src/main/java/com/okta/developer/service/dto/UserDTO.java
  * src/main/java/com/okta/developer/service/dto/package-info.java
  * src/main/java/com/okta/developer/service/mapper/UserMapper.java
  * src/main/java/com/okta/developer/service/mapper/package-info.java
  * src/main/java/com/okta/developer/service/package-info.java
  * src/main/java/com/okta/developer/web/rest/AccountResource.java
  * src/main/java/com/okta/developer/web/rest/AuditResource.java
  * src/main/java/com/okta/developer/web/rest/LogsResource.java
  * src/main/java/com/okta/developer/web/rest/UserResource.java
  * src/main/java/com/okta/developer/web/rest/errors/BadRequestAlertException.java
  * src/main/java/com/okta/developer/web/rest/errors/CustomParameterizedException.java
  * src/main/java/com/okta/developer/web/rest/errors/EmailAlreadyUsedException.java
  * src/main/java/com/okta/developer/web/rest/errors/EmailNotFoundException.java
  * src/main/java/com/okta/developer/web/rest/errors/ErrorConstants.java
  * src/main/java/com/okta/developer/web/rest/errors/ExceptionTranslator.java
  * src/main/java/com/okta/developer/web/rest/errors/FieldErrorVM.java
  * src/main/java/com/okta/developer/web/rest/errors/InternalServerErrorException.java
  * src/main/java/com/okta/developer/web/rest/errors/InvalidPasswordException.java
  * src/main/java/com/okta/developer/web/rest/errors/LoginAlreadyUsedException.java
  * src/main/java/com/okta/developer/web/rest/errors/package-info.java
  * src/main/java/com/okta/developer/web/rest/package-info.java
  * src/main/java/com/okta/developer/web/rest/util/HeaderUtil.java
  * src/main/java/com/okta/developer/web/rest/util/PaginationUtil.java
  * src/main/java/com/okta/developer/web/rest/vm/LoggerVM.java
  * src/main/java/com/okta/developer/web/rest/vm/ManagedUserVM.java
  * src/main/java/com/okta/developer/web/rest/vm/package-info.java
  * src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml
  * src/main/resources/config/liquibase/master.xml
  * src/main/resources/logback-spring.xml
  * src/main/webapp/app/app.tsx
  * src/main/webapp/app/config/axios-interceptor.ts
  * src/main/webapp/app/config/constants.ts
  * src/main/webapp/app/config/devtools.tsx
  * src/main/webapp/app/config/error-middleware.ts
  * src/main/webapp/app/config/icon-loader.ts
  * src/main/webapp/app/config/logger-middleware.ts
  * src/main/webapp/app/config/notification-middleware.ts
  * src/main/webapp/app/config/store.ts
  * src/main/webapp/app/config/translation.ts
  * src/main/webapp/app/entities/index.tsx
  * src/main/webapp/app/index.tsx
  * src/main/webapp/app/modules/administration/administration.reducer.ts
  * src/main/webapp/app/modules/administration/audits/audits.tsx
  * src/main/webapp/app/modules/administration/configuration/configuration.tsx
  * src/main/webapp/app/modules/administration/docs/docs.tsx
  * src/main/webapp/app/modules/administration/health/health-modal.tsx
  * src/main/webapp/app/modules/administration/health/health.tsx
  * src/main/webapp/app/modules/administration/index.tsx
  * src/main/webapp/app/modules/administration/logs/logs.tsx
  * src/main/webapp/app/modules/administration/metrics/metrics-modal.tsx
  * src/main/webapp/app/modules/administration/metrics/metrics.tsx
  * src/main/webapp/app/modules/administration/metrics/thread-item.tsx
  * src/main/webapp/app/modules/home/home.tsx
  * src/main/webapp/app/modules/login/logout.tsx
  * src/main/webapp/app/routes.tsx
  * src/main/webapp/app/shared/auth/private-route.tsx
  * src/main/webapp/app/shared/error/error-boundary-route.tsx
  * src/main/webapp/app/shared/error/error-boundary.tsx
  * src/main/webapp/app/shared/layout/footer/footer.tsx
  * src/main/webapp/app/shared/layout/header/header-components.tsx
  * src/main/webapp/app/shared/layout/header/header.tsx
  * src/main/webapp/app/shared/layout/header/menus/account.tsx
  * src/main/webapp/app/shared/layout/header/menus/admin.tsx
  * src/main/webapp/app/shared/layout/header/menus/entities.tsx
  * src/main/webapp/app/shared/layout/header/menus/index.ts
  * src/main/webapp/app/shared/layout/header/menus/locale.tsx
  * src/main/webapp/app/shared/layout/password/password-strength-bar.tsx
  * src/main/webapp/app/shared/model/user.model.ts
  * src/main/webapp/app/shared/reducers/action-type.util.ts
  * src/main/webapp/app/shared/reducers/application-profile.ts
  * src/main/webapp/app/shared/reducers/authentication.ts
  * src/main/webapp/app/shared/reducers/index.ts
  * src/main/webapp/app/shared/reducers/locale.ts
  * src/main/webapp/app/shared/reducers/user-management.ts
  * src/main/webapp/app/shared/util/date-utils.ts
  * src/main/webapp/app/shared/util/entity-utils.ts
  * src/main/webapp/app/shared/util/pagination.constants.ts
  * src/main/webapp/app/shared/util/url-utils.ts
  * src/main/webapp/app/typings.d.ts
  * src/test/java/com/okta/developer/config/WebConfigurerTest.java
  * src/test/java/com/okta/developer/config/WebConfigurerTestController.java
  * src/test/java/com/okta/developer/repository/CustomAuditEventRepositoryIntTest.java
  * src/test/java/com/okta/developer/repository/search/UserSearchRepositoryMockConfiguration.java
  * src/test/java/com/okta/developer/security/SecurityUtilsUnitTest.java
  * src/test/java/com/okta/developer/service/UserServiceIntTest.java
  * src/test/java/com/okta/developer/web/rest/AccountResourceIntTest.java
  * src/test/java/com/okta/developer/web/rest/AuditResourceIntTest.java
  * src/test/java/com/okta/developer/web/rest/LogsResourceIntTest.java
  * src/test/java/com/okta/developer/web/rest/TestUtil.java
  * src/test/java/com/okta/developer/web/rest/UserResourceIntTest.java
  * src/test/java/com/okta/developer/web/rest/errors/ExceptionTranslatorIntTest.java
  * src/test/java/com/okta/developer/web/rest/errors/ExceptionTranslatorTestController.java
  * src/test/java/com/okta/developer/web/rest/util/PaginationUtilUnitTest.java
  * src/test/javascript/e2e/modules/account/account.spec.ts
  * src/test/javascript/e2e/modules/administration/administration.spec.ts
  * src/test/javascript/e2e/page-objects/base-component.ts
  * src/test/javascript/e2e/page-objects/navbar-page.ts
  * src/test/javascript/e2e/page-objects/signin-page.ts
  * src/test/javascript/e2e/util/utils.ts
  * src/test/javascript/jest.conf.js
  * src/test/javascript/protractor.conf.js
  * src/test/javascript/spec/app/config/axios-interceptor.spec.ts
  * src/test/javascript/spec/app/config/notification-middleware.spec.ts
  * src/test/javascript/spec/app/modules/administration/administration.reducer.spec.ts
  * src/test/javascript/spec/app/shared/auth/private-route.spec.tsx
  * src/test/javascript/spec/app/shared/error/error-boundary-route.spec.tsx
  * src/test/javascript/spec/app/shared/error/error-boundary.spec.tsx
  * src/test/javascript/spec/app/shared/layout/header/header.spec.tsx
  * src/test/javascript/spec/app/shared/layout/header/menus/account.spec.tsx
  * src/test/javascript/spec/app/shared/reducers/application-profile.spec.ts
  * src/test/javascript/spec/app/shared/reducers/authentication.spec.ts
  * src/test/javascript/spec/app/shared/reducers/locale.spec.ts
  * src/test/javascript/spec/app/shared/reducers/user-management.spec.ts
  * src/test/javascript/spec/app/shared/util/entity-utils.spec.ts
  * src/test/javascript/spec/app/utils.ts
  * src/test/javascript/spec/enzyme-setup.ts
  * src/test/javascript/spec/storage-mock.ts
  * src/test/resources/logback.xml
This may lead to missing/broken features in SonarQube

> Task :sonarqube FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sonarqube'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sonarqube'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:273)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:258)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.NullPointerException
        at org.sonarsource.scm.git.GitScmProvider.revisionId(GitScmProvider.java:104)
        at org.sonar.scanner.report.MetadataPublisher.addScmInformation(MetadataPublisher.java:128)
        at org.sonar.scanner.report.MetadataPublisher.publish(MetadataPublisher.java:100)
        at org.sonar.scanner.report.ReportPublisher.generateReportFile(ReportPublisher.java:150)
        at org.sonar.scanner.report.ReportPublisher.execute(ReportPublisher.java:131)
        at org.sonar.scanner.phases.PublishPhaseExecutor.publishReportJob(PublishPhaseExecutor.java:73)
        at org.sonar.scanner.phases.PublishPhaseExecutor.executeOnRoot(PublishPhaseExecutor.java:55)
        at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:93)
        at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:180)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
        at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:302)
        at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:297)
        at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:271)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
        at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
        at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
        at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
        at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
        at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
        at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
        at com.sun.proxy.$Proxy100.execute(Unknown Source)
        at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
        at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
        at org.sonarqube.gradle.SonarQubeTask.run(SonarQubeTask.java:99)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:786)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:753)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 31 more
```

##### **Motivation for or Use Case**

Everything should work out-of-the-box.

##### **Reproduce the error**

Create an app with the JDL above and run the same commands I did.

##### **JHipster Version(s)**

```
health-points@0.0.0 /Users/mraible/okta-react-native-spring-boot-example/jhipster-api
└── generator-jhipster@5.4.2
```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "clientPackageManager": "npm",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": [
      "en",
      "es"
    ],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.okta.developer",
    "packageFolder": "com/okta/developer",
    "prodDatabaseType": "postgresql",
    "searchEngine": "elasticsearch",
    "serviceDiscoveryType": false,
    "skipClient": false,
    "skipServer": false,
    "testFrameworks": [
      "protractor"
    ],
    "websocket": false,
    "applicationType": "monolith",
    "baseName": "HealthPoints",
    "authenticationType": "oauth2",
    "buildTool": "gradle",
    "clientFramework": "react",
    "useSass": true,
    "jhipsterVersion": "5.4.2",
    "skipUserManagement": true,
    "cacheProvider": "ehcache",
    "serverPort": "8080"
  },
  "entities": []
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

ls: no such file or directory: .jhipster/*.json
<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "1.8.0_181"
OpenJDK Runtime Environment (Zulu 8.31.0.1-macosx) (build 1.8.0_181-b02)
OpenJDK 64-Bit Server VM (Zulu 8.31.0.1-macosx) (build 25.181-b02, mixed mode)

git version 2.17.1 (Apple Git-112)

node: v8.11.3

npm: 6.4.1

yeoman: 2.0.5

yarn: 1.10.1

Docker version 18.06.1-ce, build e68fc7a

docker-compose version 1.22.0, build f46880f

**NOTE:** This also highlights another issue - creating an app with `import-jdl` doesn't create any <entity>.json files in the `.jhipster` directory.

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by atomfrede** [comment]
Strange. Will check. Do you mean after committing sonarqube also work
correctly?

**Comment by mraible** [comment]
Yes, after I did an add and commit, it worked as expected.

**Comment by mraible** [comment]
Nevermind about the empty `.jhipster` directory. That was because I didn't specify the entities in my application definition. I had to add an `entities` setting for everything to work:

```
application {
  config {...}
  entities Points, BloodPressure, Weight, Preferences 
}

**Comment by atomfrede** [comment]
I wonder what git commit has to do with sonar. Will try to find something
out. But don't we do a git commit right after generation ?

**Comment by mraible** [comment]
This part of the stacktrace is what made me suspect it was a lack of a commit:

```
Caused by: java.lang.NullPointerException
        at org.sonarsource.scm.git.GitScmProvider.revisionId(GitScmProvider.java:104)
```

**Comment by mraible** [comment]
@atomfrede It doesn't look like we do an add/commit. I just tried generating a new project and then ran `git status`. Here's the results:

```

**Comment by atomfrede** [comment]
Strange I am pretty sure I always need to unlock my pgp key for signing and
my ssh key when generating a new project. Will try it later if I find some
time.

**Comment by PierreBesson** [comment]
@atomfrede, this is when generating the app from a JDL file. Indeed I don't think it commit the code after generation. And maybe we should not do it. What about monorepos ?

**Comment by atomfrede** [comment]
:heavy_check_mark: So what I found out, if there is `.git` folder sonar assumes git as scm tool. Furthermore sonar assumes no uncommited changes in the working directory. The maven build also doesn't work when there are no commits in the repository (but it doesn't start sonar as it fails before when trying to execute the git commit id plugin).

For the gradle build you can set `property "sonar.scm.disabled", "true"` (or removing the `.git` folder) but I don't think we should do that in general. So basically I am not sure if we can do much about the sonarqube plugin as it is build in the scanner itself.

**Comment by deepu105** [comment]
There was a bug with git commit which I fixed recently

**Comment by atomfrede** [comment]
I was on the latest master I think, but let me check again later to make sure.

**Comment by atomfrede** [comment]
@deepu105 I just verified and it doesn't work. Using `jhipster jdl` doesn't commit anything. Therefore sonar doesn't work with gradle, with maven we even can't compile (`mvn clean compile -Pprod`).

**Comment by deepu105** [comment]
Ok I need to check

**Comment by PierreBesson** [comment]
For me it's not a bug. Obviously if you initialized the git repo but did no commit it fails in weird ways but this will never happen in a real use case . So apart from this it seems to be working fine. Indeed there is the other issue of the git commit not working properly when generating an app from JDL.

**Comment by jdubois** [comment]
Fixed as #8571 is merged
